src/components/views/directory/NetworkDropdown.js

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+export default class NetworkDropdown extends React.Component {
+    constructor() {
+        super();
+
+        this.dropdownRootElement = null;
+        this.ignoreEvent = null;
+
+        this.onInputClick = this.onInputClick.bind(this);
+        this.onRootClick = this.onRootClick.bind(this);
+        this.onDocumentClick = this.onDocumentClick.bind(this);
+        this.onNetworkClick = this.onNetworkClick.bind(this);
+        this.collectRoot = this.collectRoot.bind(this);
+
+        this.state = {
+            expanded: false,
+            selectedNetwork: null,
+        };
+
+        this.networks = [
+            'matrix:matrix_org',
+            'gitter',
+            'irc:freenode',
+            'irc:mozilla',
+            'irc:w3c',
+        ];
+
+        this.networkNames = {
+            'matrix:matrix_org': 'matrix.org',
+            'irc:freenode': 'Freenode',
+            'irc:mozilla': 'Mozilla',
+            'irc:w3c': 'W3C',
+            'gitter': 'Gitter',
+        };
+
+        this.networkIcons = {
+            'matrix:matrix_org': '//matrix.org/favicon.ico',
+            'irc:freenode': '//matrix.org/_matrix/media/v1/download/matrix.org/DHLHpDDgWNNejFmrewvwEAHX',
```

Comment:
hah :P it might be a bit less cute to just include the icons in vector though otherwise these are adding unnecessary 3rd party dependencies and vector is meant to be usable without 'net access.